test/plots/barPlotTests.ts

```diff
@@ -694,6 +694,41 @@ describe("Plots", () => {
       });
     });
 
+    describe("Vertical Bar Plot label visibility", () => {
+      var svg: d3.Selection<void>;
+      var plot: Plottable.Plots.Bar<number, number>;
+      var xScale: Plottable.Scales.Linear;
+      var yScale: Plottable.Scales.Linear;
+
+      beforeEach(() => {
+        svg = TestMethods.generateSVG();
+        xScale = new Plottable.Scales.Linear();
+        yScale = new Plottable.Scales.Linear();
+        plot = new Plottable.Plots.Bar<number, number>();
+        plot.x((d) => d.x, xScale);
+        plot.y((d) => d.y, yScale);
+      });
+
+      it("hides labels outside of the visible render area", () => {
+        var data = [{ x: 1, y: -10.1 }, { x: 2, y: -5.3 }, { x: 3, y: -2.8 }];
```

Comment:
xScale.domain([1, 3]);

Labels are placed in the middle of the bar, forcing the domain slices the labels in the middle.